app/params/keyword_params.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class KeywordParams < ApplicationParams
+  attr_reader :filter
+
+  def initialize(params = {})
+    super
+
+    @filter = handle_filter_param
+  end
+
+  private
+
+  attr_accessor :adwords_url_contains,
+                :result_url_contains,
+                :word,
+                :match_at_least
+
+  def handle_filter_param
+    @adwords_url_contains = handle_param(:adwords_url_contains)
+    @result_url_contains = handle_param(:result_url_contains)
+    @word = handle_param(:word)
+    @match_at_least = handle_param(:match_at_least)
+
+    if adwords_url_contains || result_url_contains
+      url_filter
+    elsif word && match_at_least
+      match_at_least_filter
+    else
+      no_filter
+    end
```

Comment:
We prefer to use the return early instead

```suggestion
    return url_filter if adwords_url_contains || result_url_contains
    return match_at_least_filterword && match_at_least

    no_filter
```